02-peer-protocol.md

```diff
@@ -157,7 +158,9 @@ node can offer.
 (i.e. 1/4 the more normally-used 'satoshi per 1000 vbytes') that this
 side will pay for commitment and HTLC transactions, as described in
 [BOLT #3](03-transactions.md#fee-calculation) (this can be adjusted
-later with an `update_fee` message).
+later with an `update_fee` message). Note that if
+`option_anchor_outputs` is negotiated, this `feerate_per_kw`
+can be set to the min relay fee.
```

Comment:
Setting `feerate_per_kw` to the minimum relay fee is dangerous because Bitcoin Core doesn't yet have package relay (including upcoming 0.19).  See previous discussion here: https://lists.linuxfoundation.org/pipermail/lightning-dev/2018-November/001666.html

If you're planning to deploy this option prior to package relay becoming available, I think you want some safety buffer, maybe something based on the minimum feerates seen during the December '17/January '18 peak feerates.

Reply:
Ok. I think this is especially important if we'd remove the `update_fee` message. Then channel opening is the only time at which the fee rate can be set. But with `update_fee` still available, it seems to be not as critical and no safety buffer is needed?

Reply:
I don't think it can safely recommend using a minimal feerate.  It isn't clear to me where this minimal feerate comes from.  If it's the Bitcoin Core default minimal feerate (currently 0.00010000/kvB), then it an attacker could fill even an empty mempool with junk almost instantly at a cost probably below 1 BTC, raising the rolling minimum feerate and making it impossible to add the commitment transaction to the mempool (or evicting it, if already present).  It'd cost the attacker about 0.01 BTC per block to keep the mempool above the default minimum feerate, preventing the commitment transaction from entering the mempool and being CPFP-eligible before any of its CLTV-based paths became spendable.

If the minimal feerate is the value obtained from your own node or from a third-party feerate tracker, there's no way to be sure that's reflective of the value other nodes are actually using at the moment, so if your node says its BIP133 feefilter is 0.0002000 BTC/kvB, it could easily be the case that the reality is that it's 0.0002100 BTC/kvB and so (again) the commitment transaction won't enter the mempool and become CPFP-eligible.

I'm assuming current LN implementations are setting feerates that have some fixed minimum value and which are updated periodically to ensure the transaction remains in the top n MB of the mempool (with *n* inversely correlated to the CLTV delta they accept).  It seems that strategy, rather than minimum fee targeting, should continue to be used until package relay is reasonably well deployed.

Reply:
Yes, that was my thought too. Keep the path open to adjust the fee once the channel is established. The `feerate_per_kw` parameter that we are talking about here is just the fee rate on the initial version of the commitment tx.

Reply:
Resolving this commit for now. Keeping `update_fee` gives us a better chance to survive a high fee environment, although it isn't fully bullet proof.